frontend/js/src/user-feed/UserFeed.tsx

```diff
@@ -207,7 +207,45 @@ export default function UserFeedPage() {
 
   const { pages } = data || {}; // safe destructuring of possibly undefined data object
   // Flatten the pages of events from the infite query
-  const events = pages?.map((page) => page.events).flat();
+  const events = pages?.map((page) => page.events).flat() ?? [];
+
+  // Events mentioned in thank you events but not part of current cache, fetched separately as needed
+  const [separatelyLoadedEvents, setSeparatelyLoadedEvents] = React.useState<TimelineEvent<EventMetadata>[]>([])
+
+  React.useEffect(() => {
+    async function fetchOGEvents(missingEventsIDs:number[]) {
+      // Prefetch each missing original event missing from thank you events
+      // separately from fetching the feed pages
+      const promises = missingEventsIDs.map( eventId => {
+        return queryClient.ensureQueryData({
+          queryKey: ['feed-event', eventId],
+          queryFn: () => APIService.getFeedEvent(eventId, currentUser.name, currentUser.auth_token as string),
+        });
+      });
+      const resultsArray:TimelineEvent<EventMetadata>[] = [];
+      const promiseResults = await Promise.allSettled(promises);
+      promiseResults.forEach((res,idx) => {
```

Comment:
Lookss like the linter is not working on this file. Otherwise there would have been a space after the comma.